app/models/manageiq/providers/vmware/infra_manager/event_catcher/runner.rb

```diff
@@ -42,27 +42,54 @@ def monitor_events
     reset_event_monitor_handle
   end
 
-  def process_event(event)
+  def event_dedup_key(event)
+    # duplicate the event but remove ids and timestamps that change with every event
+    # the remaining attributes are used to determine whether an event is a duplicate of another
+    event.except("key", "chainId", "createdTime").tap do |hash|
+      hash["info"] &&= hash["info"].except("key", "task", "queueTime", "eventChainId")
+    end
+
+    hash = event.except("key", "chainId", "createdTime")
+    info = hash["info"]
+    return hash unless info
+
+    hash["info"] = info.except("key", "task", "queueTime", "eventChainId")
+    hash
+  end
+
+  def event_dedup_descriptor(event)
+    event_dedup_key(event)
+  end
+
+  def filtered?(event)
+    event_type = event['eventType']
+    return true if event_type.nil?
+
+    sub_event_type, display_name = _parse_event(event)
+
+    return false unless filtered_events.include?(event_type) || filtered_events.include?(sub_event_type)
+
+    _log.info("#{log_prefix} Skipping caught event [#{display_name}] chainId [#{event['chainId']}]")
+    true
+  end
+
+  def queue_event(event)
+    _sub_event_type, display_name = _parse_event(event)
+    _log.info("#{log_prefix} Queueing event [#{display_name}] chainId [#{event['chainId']}]")
+    EmsEvent.add_queue('add_vc', @cfg[:ems_id], event)
+  end
+
+  def _parse_event(event)
```

Comment:
What is the leading `_` notation for? If thats for private methods, I'm not sure if we really have this convention.